tests/integration/suite/test_catalog.py

```diff
@@ -7,6 +7,20 @@
 from .conftest import set_server_version
 
 
+def wait_for_atleast_workload(pclient, nsid, timeout=60, count=0):
```

Comment:
Are you remaking wait_for just to add `count` functionality ? Could arguably add that param as optional to wait_for, or make a generic `wait_for_count` method.